packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,306 @@
+
+const AuditLogClickhouseArrayModel = z.array(AuditLogClickhouseObjectModel);
+
+@Injectable({
+  scope: Scope.Operation,
+})
+/**
+ * Responsible for accessing audit logs.
+ */
+export class AuditLogManager {
+  private logger: Logger;
+
+  constructor(
+    logger: Logger,
+    private clickHouse: ClickHouse,
+    @Inject(S3_CONFIG) private s3Config: S3Config,
+    private emailProvider: Emails,
+    private session: Session,
+  ) {
+    this.logger = logger.child({ source: 'AuditLogManager' });
+  }
+
+  async getPaginatedAuditLogs(
+    organizationSlug: string,
+    pagination?: { first: number; after: number },
+    filter?: { startDate?: Date; endDate?: Date },
+  ): Promise<{ data: AuditLogType[] }> {
+    this.logger.info(
+      'Getting audit logs (organizationId=%s, filter=%o, pagination=%o)',
+      organizationSlug,
+      filter,
+      pagination,
+    );
+
+    const sqlFirst = sql.raw(pagination?.first ? pagination.first.toString() : '25');
+    const sqlAfter = sql.raw(pagination?.after ? pagination.after.toString() : '0');
+
+    const where: SqlValue[] = [];
+    where.push(sql`organization_id = ${organizationSlug}`);
+
+    if (filter?.startDate && filter?.endDate) {
+      const from = this.formatToClickhouseDateTime(filter.startDate.toISOString());
+      const to = this.formatToClickhouseDateTime(filter.endDate.toISOString());
+      where.push(sql`event_time >= ${from} AND event_time <= ${to}`);
+    }
+    const whereClause = where.length > 0 ? sql`WHERE ${sql.join(where, ' AND ')}` : sql``;
+
+    const result = await this.clickHouse.query({
+      query: sql`
+        SELECT *
+        FROM audit_log
+        ${whereClause}
+        ORDER BY event_time DESC
+        LIMIT ${sqlFirst}
+        OFFSET ${sqlAfter}
+      `,
+      queryId: 'get-audit-logs',
+      timeout: 10000,
+    });
+
+    const data = AuditLogClickhouseArrayModel.parse(result.data);
+
+    return {
+      data,
+    };
+  }
+
+  async uploadToS3(
+    data: string,
+    key: string,
+  ): Promise<{ ok: { url: string } } | { error: { message: string } }> {
+    const s3Storage = this.s3Config[0];
+    const { endpoint, bucket, client } = s3Storage;
+
+    const uploadResult = await client.fetch([endpoint, bucket, key].join('/'), {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'text/csv',
+      },
+      body: data,
+    });
+
+    if (!uploadResult.ok) {
+      return {
+        error: {
+          message: 'Failed to upload the file',
+        },
+      };
+    }
+
+    const getPresignedUrl = await client.fetch([endpoint, bucket, key].join('/'), {
+      method: 'GET',
+      aws: {
+        signQuery: true,
+      },
+    });
+
+    if (!getPresignedUrl.ok) {
+      return {
+        error: {
+          message: 'Failed to get the pre-signed URL',
+        },
+      };
+    }
+
+    return {
+      ok: {
+        url: getPresignedUrl.url,
+      },
+    };
+  }
+
+  async exportAndSendEmail(
```

Comment:
There needs to be an authenticated access check, otherwise a user can send audit logs of any organization to their email, even though they are not the admin/member of the organization.

Reply:
Please also cover that scenario with an integration test

Reply:
TODO @TuvalSimha See above comments ^

Reply:
Done :)